File: Dev/unitTest_Engine_cpp_gtest/Graphics/2D/CameraObject2D.cpp
#include <gtest/gtest.h>
#include <Altseed.h>
#include "../../EngineTest.h"

using namespace std;
using namespace asd;

class Graphics_CameraObject2D : public EngineTest
{
public:
	Graphics_CameraObject2D(bool isOpenGLMode)
		: EngineTest(asd::ToAString("CameraObject2D"), isOpenGLMode, 60)
	{
	}

private:
	shared_ptr<CameraObject2D> m_camera;

protected:
	void OnStart()
	{
		auto scene = make_shared<Scene>();
		auto layer = make_shared<Layer2D>();
		auto obj = make_shared<TextureObject2D>();
		m_camera = make_shared<CameraObject2D>();

		obj->SetTexture(asd::Engine::GetGraphics()->CreateTexture2D(ToAString("Data/Texture/Sample1.png").c_str()));
		m_camera->SetSrc(RectI(100, 100, 312, 312));
		m_camera->SetDst(RectI(10, 10, 200, 200));

		layer->AddObject(obj);
		layer->AddObject(m_camera);
		scene->AddLayer(layer);
		asd::Engine::ChangeScene(scene);
	}

	void OnUpdating()
	{
		auto dst = m_camera->GetDst();
		dst.X += 3;
		m_camera->SetDst(dst);
	}
};

ENGINE_TEST(Graphics, CameraObject2D)

File: Dev/unitTest_Engine_cs/Graphics/2D/CameraObject2D.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace unitTest_Engine_cs.ObjectSystem2D
{
	class CameraObject2D : EngineTest
	{
		private asd.CameraObject2D camera;

		public CameraObject2D()
			: base(60)
		{
		}

		protected override void OnStart()
		{
			var scene = new asd.Scene();
			var layer = new asd.Layer2D();
			var obj = new asd.TextureObject2D();
			camera = new asd.CameraObject2D();

			obj.Texture = asd.Engine.Graphics.CreateTexture2D("Data/Texture/Sample1.png");
			camera.Src = new asd.RectI(100, 100, 312, 312);
			camera.Dst = new asd.RectI(10, 10, 200, 200);

			layer.AddObject(obj);
			layer.AddObject(camera);
			scene.AddLayer(layer);
			asd.Engine.ChangeScene(scene);
		}

		protected override void OnUpdating()
		{
			var dst = camera.Dst;
			dst.X += 3;
			camera.Dst = dst;
		}
	}
}
